packer/builder/azure/arm/deployment_factory.go

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) Microsoft Open Technologies, Inc.
+// All Rights Reserved.
+// Licensed under the Apache License, Version 2.0.
+// See License.txt in the project root for license information.
+
+package arm
+
+import (
+	"encoding/json"
+
+	"github.com/Azure/azure-sdk-for-go/arm/resources"
+)
+
+type DeploymentFactory struct {
+	template string
+}
+
+func NewDeploymentFactory(template string) DeploymentFactory {
+	return DeploymentFactory{
+		template: template,
+	}
+}
+
+func (f *DeploymentFactory) create(templateParameters TemplateParameters) (*resources.Deployment, error) {
```

Comment:
Let's wait for more code, and (hopefully) more context.  I was attempting to group the methods on the struct, and emulate behavior I'm familiar with in C#/Java.  I'm still very much a Go n00b, so I'm probably not being very idiomatic.